### PR TITLE
Add java8 option 2.3.0

### PR DESCRIPTION
**Pull request by Randgalt**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [X] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Ports #5863 to 2.3.0

### CODE REVIEW

**Comment by Randgalt** [comment]
I think I got everything right. I tested against my own swagger file and it works.

**Comment by wing328** [comment]
@Randgalt I'll review later today.

**Comment by wing328** [comment]
Tests passed via https://circleci.com/gh/swagger-api/swagger-codegen/548

File: samples/client/petstore/java/okhttp-gson-parcelableModel/src/main/java/io/swagger/client/JSON.java [hunk 1]
```diff
@@ -113,17 +114,18 @@ public String serialize(Object obj) {
...
 
-        public OffsetDateTimeTypeAdapter() {
-            this(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        public DateTimeTypeAdapter() {
+            this(ISODateTimeFormat.dateTime().withOffsetParsed());
         }
 
-        public OffsetDateTimeTypeAdapter(DateTimeFormatter formatter) {
+        public DateTimeTypeAdapter(DateTimeFormatter formatter) {
             this.formatter = formatter;
```
Comment (lwander) [inline samples/client/petstore/java/okhttp-gson-parcelableModel/src/main/java/io/swagger/client/JSON.java]: This field `formatter` was removed field in favor of `parseFormatter` and `printFormatter` in this PR. This breaks compilation of the generated client. Which field did you intend to set here @Randgalt?

Comment (Randgalt) [inline samples/client/petstore/java/okhttp-gson-parcelableModel/src/main/java/io/swagger/client/JSON.java]: This code is autogenerated so I'm not sure how to answer.

Comment (lwander) [inline samples/client/petstore/java/okhttp-gson-parcelableModel/src/main/java/io/swagger/client/JSON.java]: Ah you're right sorry, looks like you picked up a change to a template that came out a bad merge. I'll keep digging.

**Pull request merged**